README.md

```diff
@@ -1 +1,207 @@
+
+Full specification detailing the implementations can be found on [here](https://rdf-connect.github.io/specification/).
+
+## Core Capabilities
+
+### 🚀 **Distributed Processing Architecture**
+- **Orchestrator-Runner Model**: Centralized orchestration with distributed execution
+- **Processor Lifecycle Management**: Dynamic processor initialization, configuration, and lifecycle control
+- **Bidirectional Streaming**: Full-duplex communication between orchestrators and runners
+
+### 📡 **Advanced Message Routing**
+- **Channel-Based Routing**: Messages routed through logical channels with sequence numbering
+- **URI-Based Identification**: Unique component identification system
+
+### 🌊 **Streaming Data Protocol**
+- **Chunked Streaming**: Efficient streaming of large datasets with identification
+- **Stream Identification**: Each stream has unique channel and local sequence number identifiers
+- **Data Chunking**: Optimized data transmission with metadata preservation
+
+### 🔧 **Robust Error Handling**
+- **Built-in Error Reporting**: Comprehensive error handling throughout the protocol
+- **Graceful Degradation**: Optional error fields allow for partial success scenarios
+- **Service-Level Error Propagation**: Errors bubble up through the service interface
+
+### 📊 **Pipeline Management**
+- **Turtle Format Support**: Full pipeline information in Turtle RDF serialization
+- **SHACL Integration**: Built-in support for SHACL (Shapes Constraint Language) validation
+- **Dynamic Pipeline Updates**: Runtime pipeline modification capabilities
+
+### 📝 **Comprehensive Logging**
+- **Dedicated Log Streams**: Separate logging channels for debugging and monitoring
+- **Structured Logging**: Log messages with levels, entities, and aliases
+- **Real-time Log Streaming**: Continuous log streaming for live monitoring
+
+## Protocol Components
+
+### Core Message Types
+
+#### `common.proto` - Base Protocol Definitions
+- `Error`: Error reporting with cause description
+- `Close`: Channel closure signaling
+- `DataChunk`: Raw data chunk for streaming
+- `SendingMessage`: Message sent from runner to orchestrator with channel, data, and local sequence number
+- `ReceivingMessage`: Message received by runner from orchestrator with global sequence number, channel, and data
+- `ReceivingStreamMessage`: Stream message received by runner with global sequence number and channel
+- `ReceivingStreamControl`: Control message for streaming to producer (contains stream sequence number)
+- `SendingStreamControl`: Control message for streaming from consumer (contains global or stream sequence number)
+- `StreamIdentify`: First message when runner sends streaming data (channel, local sequence number, runner URI)
+- `StreamChunk`: Wrapper for StreamIdentify or DataChunk messages
+- `MessageProcessed`: Processing acknowledgment with channel and global sequence number
```

Comment:
I think this one will need to be updated to reflect the `GlobalAck`, `LocalAck` changes.